refinery/core/views.py

```diff
@@ -1015,11 +1015,82 @@ def partial_update(self, request, uuid, format=None):
 
         return Response(serializer.data)
 
-    def is_user_a_group_manager(self, user, group):
+
+class GroupMemberAPIView(APIView):
+    """API endpoint that allows for Group Members to be promoted,
+    demoted or removed"""
+    http_method_names = ['delete', 'post']
+
+    def get_object(self, uuid):
+        try:
+            return ExtendedGroup.objects.get(uuid=uuid)
+        except ExtendedGroup.DoesNotExist as e:
+            logger.error(e)
+            raise Http404
+        except ExtendedGroup.MultipleObjectsReturned as e:
+            logger.error(e)
+            raise APIException("Multiple groups returned for this request.")
+
+    def get_user(self, id):
+        try:
+            return User.objects.get(id=id)
+        except User.DoesNotExist as e:
+            logger.error(e)
+            raise Http404
+        except User.MultipleObjectsReturned as e:
+            logger.error(e)
+            raise APIException("Multiple users returned for this request.")
+
+    def post(self, request, uuid):
+        # if group is a manager_group, user is promoted by adding to user set
+        group = self.get_object(uuid)
+
+        if group.is_user_a_group_manager(request.user):
+            edit_user = self.get_user(request.data.get('userId'))
+            if group.is_manager_group():
+                group.user_set.add(edit_user)
+                serializer = ExtendedGroupSerializer(group,
+                                                     context={user: edit_user})
+                return Response(serializer.data)
+            return HttpResponseBadRequest(
+                content="Manager groups are required to upgrade."
+                )
+        # not a group manager
+        return Response(uuid, status=status.HTTP_403_FORBIDDEN)
+
+    def delete(self, request, uuid, id):
+        # if group is a manager_group, user is demoted by removal
+        group = self.get_object(uuid)
+        edit_user = self.get_user(id)
+        if self.is_user_authorized_to_edit(group, request.user, edit_user):
+            return Response(uuid, status=status.HTTP_403_FORBIDDEN)
```

Comment:
If user is authorized to edit then return HTTP 403?

Reply:
Forgot the un, will update the method name.

Reply:
I'd rather keep the name and simplify the logic.